sherpa/astro/ui/tests/test_astro_ui_utils_simulation.py

```diff
@@ -564,3 +565,186 @@ def test_fake_pha_issue_1568(make_data_path, clean_astro_ui, reset_seed):
+
+    arf, rmf = setup_fake_pha_test()
+    ui.fake_pha(1, arf, rmf, 100, bkg=None)
+
+    # Check what's changed and what hasn't
+    ui.set_analysis(1, "energy", type="counts")
+    assert ui.get_data_plot().y == pytest.approx(expected)
+    assert ui.get_bkg_plot().y == pytest.approx(bexpected)
+    assert ui.get_model_plot().y == pytest.approx([80, 80, 300])
+    assert ui.get_bkg_model_plot().y == pytest.approx([0, 0, 1200])
+
+    assert ui.get_source_plot().y == pytest.approx([2, 1, 0])
+    assert ui.get_bkg_source_plot().y == pytest.approx([0, 0, 3])
+
+    ui.set_analysis(1, "energy", type="rate")
+    assert ui.get_data_plot().y == pytest.approx(expected / np.asarray([20, 40, 100]))
+    assert ui.get_bkg_plot().y == pytest.approx(bexpected / np.asarray([100, 200, 500]))
+    assert ui.get_model_plot().y == pytest.approx([4, 2, 3])
+    assert ui.get_bkg_model_plot().y == pytest.approx([0, 0, 2.4])
+
+    assert ui.get_source_plot().y == pytest.approx([2, 1, 0])
+    assert ui.get_bkg_source_plot().y == pytest.approx([0, 0, 3])
+
+
+def test_fake_pha_with_bgnd_model(clean_astro_ui, reset_seed):
+    """Add a test found when investigating #1685
+
+    At some point working on #1684 this included the background twice,
+    so add the test in to ensure this never happens.
+
+    """
+
+    expected = [85, 76, 297]
+    bexpected = [300, 300, 300]
+
+    arf, rmf = setup_fake_pha_test()
+    ui.fake_pha(1, arf, rmf, 100, bkg="model")
+
+    # Check what's changed and what hasn't
+    ui.set_analysis(1, "energy", type="counts")
+    assert ui.get_data_plot().y == pytest.approx(expected)
+    assert ui.get_bkg_plot().y == pytest.approx(bexpected)
+    assert ui.get_model_plot().y == pytest.approx([80, 80, 300])
+    assert ui.get_bkg_model_plot().y == pytest.approx([0, 0, 1200])
+
+    assert ui.get_source_plot().y == pytest.approx([2, 1, 0])
+    assert ui.get_bkg_source_plot().y == pytest.approx([0, 0, 3])
+
+    ui.set_analysis(1, "energy", type="rate")
+    assert ui.get_data_plot().y == pytest.approx(expected / np.asarray([20, 40, 100]))
+    assert ui.get_bkg_plot().y == pytest.approx(bexpected / np.asarray([100, 200, 500]))
+    assert ui.get_model_plot().y == pytest.approx([4, 2, 3])
+    assert ui.get_bkg_model_plot().y == pytest.approx([0, 0, 2.4])
+
+    assert ui.get_source_plot().y == pytest.approx([2, 1, 0])
+    assert ui.get_bkg_source_plot().y == pytest.approx([0, 0, 3])
+
+
+def test_fake_pha_with_bgnd_data(clean_astro_ui, reset_seed):
+    """Extend test_fake_pha_with_bgnd_model to check with a dataset."""
+
+    expected = [90, 89, 1196]
+    bexpected = [10, 15, 5]
+
+    # Choose a different background to the actual background,
+    # particularly with different exposure/backscal.
+    #
+    bkg = ui.DataPHA("fakey", [1, 2, 3], bexpected)
+    bkg.exposure = 400
+    bkg.backscal = 0.1
+
+    arf, rmf = setup_fake_pha_test()
+    ui.fake_pha(1, arf, rmf, 100, bkg=bkg)
+
+    # Check what's changed and what hasn't
```

Comment:
Would it be easier to understand if the assert lines that match the previous test where marked in some way or pulled into a separate function `def fake_pha_with_bgnd_data_XXX` that would be called in both tests to make it obvious which of the lines are the same?

Reply:
I've made a change which might not be quite what you were thinking but does improve things a bit.

Reply:
This is good. I usually don't care too much about DRY in tests, but I thought it might help future me (or future you) in this particular case.